pkg/cli/newapp/newapp.go

```diff
@@ -269,7 +269,8 @@ func NewCmdNewApplication(f kcmdutil.Factory, streams genericclioptions.IOStream
 	cmd.Flags().StringSliceVar(&o.Config.SourceRepositories, "code", o.Config.SourceRepositories, "Source code to use to build this application.")
 	cmd.Flags().StringVar(&o.Config.ContextDir, "context-dir", o.Config.ContextDir, "Context directory to be used for the build.")
 	cmd.Flags().StringSliceVarP(&o.Config.ImageStreams, "image-stream", "i", o.Config.ImageStreams, "Name of an image stream to use in the app.")
-	cmd.Flags().StringSliceVar(&o.Config.DockerImages, "docker-image", o.Config.DockerImages, "Name of a Docker image to include in the app.  Note:  not specifying a registry or repository means defaults in place for client image pulls are employed.")
+	cmd.Flags().StringSliceVar(&o.Config.DockerImages, "image", o.Config.DockerImages, "Name of a container image to include in the app.  Note:  not specifying a registry or repository means defaults in place for client image pulls are employed.")
```

Comment:
@rhatdan we can't drop existing command line flags.  If you want to mark the existing flag deprecated and introduce its replacement, we can do that.

Reply:
ah, I see you aliased it below

Reply:
Aliasing isn't sufficient since it hides the problem. We want to force people to move off of using these flags. That's why we should deprecate the old ones and introduce new ones instead. In 3 releases or a year we'll be able to drop the old ones. Make sure to add a note in the release notes issue in https://github.com/openshift/openshift-docs/issues/33497 so that this is being tracked. 

From my conversation with @bparees since it might not be clear, marking a flag deprecated will effectively make it disappear from everywhere (docs, completions, et.) so you either need to know the name or have a pre-existing flag. The benefit of marking it deprecated is that you're getting a warning, to switch to new flag.